src/pages/version/BuildVariantCard/VariantTaskGroup/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { wordBreakCss, StyledRouterLink } from "components/styles";
+import { getVersionRoute } from "constants/routes";
+import { mapUmbrellaStatusToQueryParam } from "constants/task";
+import { size } from "constants/tokens";
+import { StatusCount } from "gql/generated/types";
+import { useQueryParam } from "hooks/useQueryParam";
+import { PatchTasksQueryParams, TaskStatus } from "types/task";
+import { string, statuses } from "utils";
+import { arraySymmetricDifference } from "utils/array";
+
+const { groupStatusesByUmbrellaStatus } = statuses;
+const { applyStrictRegex } = string;
+
+interface VariantTaskGroupProps {
+  displayName: string;
+  statusCounts: StatusCount[];
+  variant: string;
+  versionId: string;
+}
+const VariantTaskGroup: React.VFC<VariantTaskGroupProps> = ({
+  displayName,
+  statusCounts,
+  variant,
+  versionId,
+}) => {
+  const { sendEvent } = useVersionAnalytics(versionId);
+
+  const [variantSearch] = useQueryParam<string | null>(
+    PatchTasksQueryParams.Variant,
+    undefined
+  );
+  const [sorts] = useQueryParam(PatchTasksQueryParams.Sorts, undefined);
+  const [statusSearch] = useQueryParam<string[] | null>(
+    PatchTasksQueryParams.Statuses,
+    []
+  );
+  const hasStatusFilter = statusSearch.length > 0;
+  const hasVariantFilter = variantSearch !== undefined;
+  const isVariantSelected = variantSearch === applyStrictRegex(variant);
+  const hasAnyStatusOrVariantFilters = hasVariantFilter || hasStatusFilter;
+
+  const callBack = (taskSquareStatuses: string[]) => () => {
+    sendEvent({
+      name: "Click Grouped Task Square",
+      taskSquareStatuses,
+    });
+  };
+
+  const { stats } = groupStatusesByUmbrellaStatus(statusCounts ?? []);
+  const versionRouteParams = {
+    sorts,
+    page: 0,
+    variant: isVariantSelected ? undefined : applyStrictRegex(variant),
+  };
+  return (
+    <div data-cy="patch-build-variant">
+      <StyledRouterLink
+        css={wordBreakCss}
+        to={getVersionRoute(versionId, {
+          ...versionRouteParams,
+        })}
+        onClick={() =>
+          sendEvent({
+            name: "Click Build Variant Grid Link",
+          })
+        }
+        data-cy="build-variant-display-name"
+      >
+        {displayName}
+      </StyledRouterLink>
+
+      <TaskBadgeContainer>
+        {stats.map(
+          ({ umbrellaStatus, count, statusCounts: groupedStatusCounts }) => {
+            // If the variant is selected and there are no status filters, the badge should be active.
+            // If the variant is selected and there are status filters, the badge should be active if the umbrella status is set.
+            // If the variant is not selected, the badge should be active if the umbrella status is set.
+            const isBadgeActive =
+              (isVariantSelected && !hasStatusFilter) ||
+              (isVariantSelected &&
+                hasStatusFilter &&
+                isUmbrellaStatusSet(umbrellaStatus, statusSearch));
```

Comment:
The comments seem to be referring to three different conditions but there's only two conditions separated by `||`. Is there another condition that needs to be added (or update comment)?

Reply:
Refactored this to make it more clear